lib/indexr.js

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+/**
+ * Recursively creates index.js files exporting all *.js files in each directory.
+ */
+const path = require('path')
+const Promise = require('bluebird')
+const del = require('del')
+const yargs = require('yargs')
+
+const fs = Promise.promisifyAll(require('fs'))
+const glob = Promise.promisify(require('glob'))
+
+const argv = yargs
+  .usage('$0 <directory> [options]')
+  .example('$0 src', 'Recursively creates index.js in src')
+  .demand(1)
+  .help('help')
+  .alias('h', 'help')
+  .argv
+
+const FLAG = '/* @indexr */'
+const HEADER = [
+  FLAG,
+  '/* eslint-disable */',
+  '',
+].join('\n')
+
+const findDirs = (dir) => glob(`${dir}/**/`, {
+  ignore: [
+    `./node_modules/**`,
+  ],
+})
+
+const findFiles = (dir) => glob(`${dir}/*.js`, {
+  ignore: [
+    '**/index.js',
+  ],
+})
+
+// resolve if missing index or existing index contains FLAG
+const getSafeIndexPath = (dir) => {
+  const indexPath = path.resolve(dir, 'index.js')
+  return fs.readFileAsync(indexPath, 'utf8')
+    .then(data => data.includes(FLAG) ? indexPath : null)
+    .catch(err => err && err.code === 'ENOENT' ? indexPath : null)
+}
+
+const createIndexFile = (dir) => {
+  return Promise.all([
+      findFiles(dir),
+      getSafeIndexPath(dir),
+    ])
+    .then(all => {
+      const files = all[0]
+      const safeIndexPath = all[1]
+
+      // skip if no safe index was found
+      if (!safeIndexPath) return Promise.resolve()
+
+      // remove abandoned index
+      if (safeIndexPath && !files.length) {
+        return del(safeIndexPath)
+      }
```

Comment:
Clean up an indexr `index.js` file if there are not more `*.js` files in the directory.